### PR TITLE
Added a 'testmode' for mocked login

### PR DESCRIPTION
**Pull request by Pomax**
:magic:

### CODE REVIEW

File: testmode.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  var username = 'testuser';
+  var usermail = 'test@example.org';
+  var hash = '0c17bf66e649070167701d2d3cd71711';
+
+  var okay = {
+    status: 'okay'
+  };
+
+  var basicuser = {
```
Comment (Pomax) [inline testmode.js]: actually not used anymore, let's take this out

File: README.md [hunk 2]
```diff
@@ -46,7 +46,11 @@ var webmakerAuth = new WebmakerAuth({
   forceSSL: process.env.FORCE_SSL // default false
 
   // if a cookie is older than the given time (in milliseconds), refresh the userdata
-  refreshTime: 1000 * 60 * 5 // default 15 minutes
+  refreshTime: 1000 * 60 * 5 // default 15 minutes,
+
+  // if set to 'true', webmaker-auth will bypass true login and simply treat any attempt
+  // to log in as successful, yielding a session for user "testuser" with email "test@example.org"
+  testmode: false
```
Comment (Pomax) [inline README.md]: for naming consistency, `testMode`

File: index.js [hunk 3]
```diff
@@ -146,7 +146,7 @@ module.exports = function (options) {
     return req.connection.remoteAddress;
   }
 
-  self.handlers = {
+  self.handlers = options.testmode ? require('./testmode')(options) : {
```
Comment (Pomax) [inline index.js]: `testMode`

**Comment by cadecairos** [comment]
arrr +

**Pull request merged**